File: Models/Encadreur.cs
namespace StageursApp.Models;

public class Encadreur
{
    public int Id { get; set; }
    public string? Nom { get; set; }
    public string? Prenom { get; set; }
    public string? Email { get; set; }
    public string? Fonction { get; set; }
    public string? Telephone { get; set; }

    // Navigation
    public ICollection<Etudiant>? Etudiants { get; set; }
}

File: stageurs-laravel/app/Models/Encadreur.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Encadreur extends Model
{
    use HasFactory;

    protected $fillable = [
        'nom',
        'prenom',
        'email',
        'fonction',
        'telephone'
    ];

    /**
     * Get the students supervised by this encadreur
     */
    public function etudiants()
    {
        return $this->hasMany(Etudiant::class);
    }

    /**
     * Get the full name of the encadreur
     */
    public function getFullNameAttribute()
    {
        return $this->nom . ' ' . $this->prenom;
    }
}
